Xamarin.Essentials/SecureStorage/SecureStorage.android.cs

```diff
@@ -14,14 +14,42 @@ namespace Xamarin.Essentials
 {
     public static partial class SecureStorage
     {
+        public static bool LegacyKeyHashFallback { get; set; } = true;
+
         static readonly object locker = new object();
 
         static Task<string> PlatformGetAsync(string key)
         {
             var context = Platform.AppContext;
 
             string defaultEncStr = null;
-            var encStr = Preferences.Get(Utils.Md5Hash(key), defaultEncStr, Alias);
+
+            var keyHash = SecureStorage.Crc64(key);
+
+            if (LegacyKeyHashFallback)
+            {
+                // If not found, could have been previously stored with md5 key
+                if (!Preferences.ContainsKey(keyHash, Alias))
+                {
+                    // If previously stored with md5 key, save with crc key
+                    var md5Key = Utils.Md5Hash(key);
+                    if (Preferences.ContainsKey(md5Key, Alias))
+                    {
+                        var v = Preferences.Get(md5Key, defaultEncStr, Alias);
+                        Preferences.Set(keyHash, v, Alias);
+
+                        try
+                        {
+                            Preferences.Remove(md5Key, Alias);
```

Comment:
Do we want to read and write this many times in the get? It is not bad, but what is the issue with just reading the new, and then falling back to the old? Let the "set" methods clean up.

Reply:
My concern is if you never Set the value again you never get migrated.  Trying to be proactive and get data moved over for when we eventually drop this.

Reply:
We decided to keep this to get data migrating sooner.